compiler/rustc_next_trait_solver/src/solve/eval_ctxt/mod.rs

```diff
@@ -974,7 +974,9 @@ where
         if cfg!(debug_assertions) {
             for g in goals.iter() {
                 match g.predicate.kind().skip_binder() {
-                    ty::PredicateKind::Subtype { .. } | ty::PredicateKind::AliasRelate(..) => {}
+                    ty::PredicateKind::Subtype { .. }
+                    | ty::PredicateKind::AliasRelate(..)
+                    | ty::PredicateKind::Clause(ty::ClauseKind::WellFormed(_)) => {}
```

Comment:
`WellFormed` is required to constrain bivariant parameters

idk whether this should be productive, cc https://github.com/rust-lang/trait-system-refactor-initiative/issues/169

can you add a `FIXME(-Znext-solver=coinductive): should these WF goals also be unproductive?`. We could also just manually add each goal and use `Misc` for the WF goals 🤷 

doesn't matter for coherence rn

r=me with either approach

Reply:
> WellFormed is required to constrain bivariant parameters

yea, that's my understanding when turned this into a test.

I'll make it misc so we don't accidentally commit to this.